doozerlib/cli/release_gen_payload.py

```diff
@@ -0,0 +1,334 @@
+
+    for key in mirroring:
+        private = key.endswith("-priv")
+        arch = key[:-5] if private else key  # strip `-priv` suffix
+
+        mirror_filename = 'src_dest.{}'.format(key)
+        imagestream_filename = 'image_stream.{}'.format(key)
+        target_is_name = is_name
+        target_is_namespace = is_namespace
+        if arch != 'x86_64':
+            target_is_name = '{}-{}'.format(target_is_name, arch)
+            target_is_namespace = '{}-{}'.format(target_is_namespace, arch)
+
+        if private:
+            target_is_namespace += "-priv"  # should we also append `-priv` suffix to target_is_name?
+
+        def build_dest_name(tag_name):
+            entry = mirroring[key][tag_name]
+            if arch == 'x86_64':
+                arch_ext = ''
+            else:
+                arch_ext = '-{}'.format(arch)
+            return 'quay.io/{orgrepo}:{version}-{release}{arch_ext}-ose-{tag_name}-{uuid}'.format(orgrepo=orgrepo,
+                                                                                                  version=entry['version'],
+                                                                                                  release=entry['release'],
+                                                                                                  arch_ext=arch_ext,
+                                                                                                  tag_name=tag_name,
+                                                                                                  uuid=runtime.uuid)
+
+        # Save the default SRC=DEST 'oc image mirror' input to a file for
+        # later.
+        with io.open(mirror_filename, 'w+', encoding="utf-8") as out_file:
+            for tag_name in mirroring[key]:
+                dest = build_dest_name(tag_name)
+                out_file.write("{}={}\n".format(mirroring[key][tag_name]['image_src'], dest))
+
+        with io.open("{}.yaml".format(imagestream_filename), 'w+', encoding="utf-8") as out_file:
+            # Add a tag spec to the image stream. The name of each tag
+            # spec does not include the 'ose-' prefix. This keeps them
+            # consistent between OKD and OCP
+
+            # Template Base Image Stream object.
+            tag_list = []
+            isb = {
+                'kind': 'ImageStream',
+                'apiVersion': 'image.openshift.io/v1',
+                'metadata': {
+                    'name': target_is_name,
+                    'namespace': target_is_namespace,
+                },
+                'spec': {
+                    'tags': tag_list,
+                }
+            }
+
+            for tag_name in mirroring[key]:
+                tag_list.append({
+                    'name': tag_name,
+                    'from': {
+                        'kind': 'DockerImage',
+                        'name': build_dest_name(tag_name)
```

Comment:
The `sha-{image_sha}` convention suggested above might best be incorporated into the mirroring structure.